src/include/denc.h

```diff
@@ -1876,6 +1876,14 @@ inline std::enable_if_t<traits::supported && !traits::featured> decode_nohead(
   static_assert(CEPH_RELEASE >= (CEPH_RELEASE_SQUID /*19*/ + 2) || compat == 2);	\
   _denc_start(p, &struct_v, &struct_compat, &_denc_pchar, &_denc_u32);	\
   do {
+  
```

Comment:
// This variant is unsafe, because older versions will not even catch incompatibility.
// The ability to decode must be verified by other means,